tests/coordinate/coordinate_tests.cpp

```diff
@@ -0,0 +1,155 @@
+//  Copyright (c) 2014-2020 Andrey Upadyshev <oliora@gmail.com>
+//
+//  Use, modification and distribution are subject to the
+//  Boost Software License, Version 1.0. (See accompanying file
+//  LICENSE_1_0.txt or copy at http://www.boost.org/LICENSE_1_0.txt)
+
+#include <catch/catch.hpp>
+
+#include "ppconsul/coordinate.h"
+#include "ppconsul/agent.h"
+#include "test_consul.h"
+
+#include <algorithm>
+
+using namespace ppconsul::coordinate;
+using ppconsul::agent::Agent;
+
+namespace {
+    const auto Non_Existing_Node_Name = "D0087276-8F85-4612-AC88-8871DB15B2A7";
+
+    // Verifies that all nodes have the same dimensionality
+    bool sameDim(const std::vector<Node>& nodes)
+    {
+        if (nodes.empty())
+        {
+            return true;
+        }
+
+        auto dim = nodes.front().coord.vec.size();
+        
+        return std::none_of(
+            nodes.begin(), nodes.end(),
+            [dim](const auto& node)
+            {
+                return (node.coord.vec.size() != dim);
+            }
+        );
```

Comment:
To remove the double negation (it's hard to understand) use `std::all_of` with "positive" predicate.